src/index.js

```diff
@@ -0,0 +1,127 @@
+  document.getElementById('todoListInput').value = '';
+};
+
+// interactivity begins here
+const toggleCheckbox = (id) => {
+  const todoListArr = getDataFromLocalStorage();
+  const checkboxElement = document.getElementById(id).checked;
+  const arrIndex = todoListArr.findIndex((item) => `checkbox_${item.id}` === id);
+  todoListArr[arrIndex].completed = checkboxElement;
+  storeItem(todoListArr);
+  updateView();
+};
+
+const toggleEdit = (id, element) => {
+  const todoListArr = getDataFromLocalStorage();
+  const arrIndex = todoListArr.findIndex((item) => `${element}_${item.id}` === id);
+  todoListArr[arrIndex].editable = !todoListArr[arrIndex].editable;
+  storeItem(todoListArr);
+  updateView();
+};
+
+const saveEdit = (id, element) => {
+  const newInputValue = document.getElementById(`input_${id.split('_')[1]}`);
+  const todoListArr = getDataFromLocalStorage();
+  const arrIndex = todoListArr.findIndex((item) => `${element}_${item.id}` === id);
+  todoListArr[arrIndex].description = newInputValue.value;
+  storeItem(todoListArr);
+  toggleEdit(`toggleMode_${id.split('_')[1]}`, 'toggleMode');
+};
+
+const deleteItem = (id, element) => {
+  const todoListArr = getDataFromLocalStorage();
+  const arrIndex = todoListArr.findIndex((item) => `${element}_${item.id}` === id);
+  todoListArr.splice(arrIndex, 1);
+
+  for (let i = 0; i < todoListArr.length; i += 1) {
+    todoListArr[i].index = i + 1;
+  }
+  storeItem(todoListArr);
+  updateView();
+};
+
+const addItem = (data) => {
+  const item = {
+    completed: false,
+    description: data,
+    id: Math.random().toString(16).slice(2),
+    editable: false,
+    index: 1,
+  };
+  const previousTodoList = getDataFromLocalStorage();
+  let todoList = [];
+  if (previousTodoList !== null) {
+    item.index = previousTodoList.length + 1;
+    todoList = [...getDataFromLocalStorage(), item];
+  } else {
+    todoList.push(item);
+  }
+  storeItem(todoList);
+  clearInput();
+  updateView();
+};
+
+document.addEventListener('DOMContentLoaded', () => {
+  document.querySelector('form').addEventListener('submit', (e) => {
+    e.preventDefault();
+    const formData = new FormData(e.target);
+    addItem(formData.get('new-todo'));
+  });
+
+  document
+    .getElementById('todo-list')
+    .addEventListener('click', (e) => {
+      if (e.target.type === 'checkbox') {
+        toggleCheckbox(e.target.id);
+      } else if (e.target.type === 'button') {
+        const eventType = String(e.target.id);
+        if (eventType.indexOf('save') !== -1) {
+          saveEdit(e.target.id, 'save');
+        }
+        if (eventType.indexOf('delete') !== -1) {
+          deleteItem(e.target.id, 'delete');
+        }
+        if (eventType.indexOf('toggleMode') !== -1) {
+          toggleEdit(e.target.id, 'toggleMode');
+        }
+      }
+    });
+
+  document
+    .getElementById('clear-completed-button')
+    .addEventListener('click', () => {
+      clearLocalStorage();
+    });
+});
```

Comment:
- [ ] When I try to delete `clear all completed` tasks it throws an error and nothing is deleted so please fix that.

![image](https://user-images.githubusercontent.com/68796232/166898132-535eb543-f27a-43a1-aae6-2f87d6c98ecf.png)

**[hint]: you can fix that by implement `array.filter` method that filters when `completed == true`**